### PR TITLE
Add Frontend and Backend Structure

### PR DESCRIPTION
**Pull request by Zhangjason308**
- Create the client directory using vite, which uses svelte framework and js
- install axios as a dependency for api communication
- create backend to frontend proxy communication by modifying vite config

### CODE REVIEW

File: client/src/app.css [hunk 1]
```diff
@@ -0,0 +1,79 @@
+:root {
```
Comment (Triguy12) [inline client/src/app.css:1]: is it necessary to have a .css file when the styling can be done through a svelte component?
If global styling was the goal, the global prefix can be used in the svelte component.

File: client/src/vite-env.d.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/// <reference types="svelte" />
```
Comment (Triguy12) [inline client/src/vite-env.d.ts:1]: is this file necessary?

File: client/src/app.css (hunk 1 above)
Comment (Zhangjason308) [reply client/src/app.css:1]: I think generally you want want to have a root css file to handle most of the styling so that you dont have to add the styling to every component. We can remove it later on if needed

Comment (Triguy12) [reply client/src/app.css:1]: sounds good!

File: client/src/main.js
Comment (nicholasyung1) [inline client/src/main.js:1]: What does this file do? I assume it is something to do with the Svelte setup?

**Review by Triguy12 — APPROVED** [review]
Seems like this is mostly necessary for project setup, be sure to remove template html and scripts, and do not to include excess files in the final build.

File: client/src/main.js
Comment (Zhangjason308) [reply client/src/main.js:1]: yes it is required for the root js file

**Review by nicholasyung1 — APPROVED** [review]

**Review by Waluiyeegi — APPROVED** [review]
Looks Good!

**Review by Nervith — APPROVED** [review]
Setups looks complete. Well add missing stuff when we need it
